ARCs/arc-0014.md

```diff
@@ -0,0 +1,247 @@
+> Comments like this are non-normative.
+
+Interfaces are defined in TypeScript. All the objects that are defined are valid JSON objects, and all JSON string types are UTF-8 encoded.
+
+This ARC uses interchangeably the terms "*blockchain address*", "*public key*", and "*PK*" to indicate the on-chain address of a blockchain identity, and in particular of an Algorand account.
+
+### Overview
+
+This document describes a standard approach to authenticate Users with their blockchain identity. In particular, an Algorand address is used as a *unique identifier*, and the secret key of that address is used to digitally sign a *message* as a proof of identity for a  Verifier.
+
+To sum up, given an Algorand account <*PKa, SKa*>, this ARC defines the standards for:
+
+- create an [ARC-14](./arc-0014.md) compliant digital signature with *SKa*;
+- verify an ARC-14 compliant digital signature with *PKa*.
+
+### Assumptions
+
+The standard proposed in this document works under the following assumptions:
+
+- User and Verifier communicates over secure SSL/TLS encrypted channels;
+- The Verifier knows the Users’ *PKa*;
+- For each *PKa* the Verifier generates a unique message to be signed;
+- The message changes for each authentication request - avoid replay attack;
+- Users are the only custodians of the *SKa*;
+- The secret keys are stored into a wallet;
+- Users won’t use different *PKa* for authentication;
+- Users won’t change their public address for authentication;
+- Users **MUST** use Algorand compliant keys to sign the messages;
+- Multisig and LogicSig are not supported;
+- Users **MUST NOT** use rekeyed accounts.
+
+> Rekey breaks ARC-14 security. Applications with high security requirements **MUST** check that Users' accounts have not been rekeyed.
+
+### Authentication Mechanism
+
+The authentication mechanism defined in this ARC works as follows: a User communicates to the Verifier their intention to authenticate with their Algorand account <*PKa, SKa*>.
+
+> Algorand transforms traditional 32-bytes cryptographic keys into more readable and user-friendly objects. A detailed description of such a transformation can be found in the <a href="https://developer.algorand.org/docs/get-details/accounts/#keys-and-addresses">developer portal</a>.
+
+The Verifier requests the signing of a random message, that is required to be signed with the secret key *SKa* associated with the User's Algorand account. The User generates the digital signature using a Wallet, and sends it back to the Verifier. At that stage, the Wallet **MUST** check that the User is signing the message for the correct Verifier (to protect Users against man-in-the-middle attack). Finally, the Verifier verifies the signature and, if everything is good, authenticates the User.
```

Comment:
For posterity: now that the scope will be limited to online scenarios for specific PKs (eg: known current authaddr), then at this point no message need be generated and nothing need be signed. Bob can construct an ECIES message for Alice with a nonce. Asking Alice to sign things introduces security threats of various kinds.